fe/fe-core/src/main/java/com/starrocks/catalog/AggregateFunction.java

```diff
@@ -330,6 +332,21 @@ public String toSql(boolean ifNotExists) {
         return sb.toString();
     }
 
+    @Override
+    public boolean equals(Object obj) {
+        if (!(obj instanceof AggregateFunction)) {
+            return false;
+        }
+        AggregateFunction agg = (AggregateFunction) obj;
+        if (!Objects.equals(intermediateType, agg.intermediateType) || ignoresDistinct != agg.ignoresDistinct ||
+                isAnalyticFn != agg.isAggregateFn || returnsNonNullOnEmpty != agg.returnsNonNullOnEmpty ||
+                !Objects.equals(symbolName, agg.symbolName) || isDistinct != agg.isDistinct ||
+                !Objects.equals(nullsFirst, agg.getNullsFirst()) || !Objects.equals(isAscOrder, agg.getIsAscOrder())) {
+            return false;
+        }
+        return super.equals(obj);
```

Comment:
It seems not easy to read.  How about firstly satisfied `super.equals(obj)` then do the private field comparing process.